transcom/mymove
# Am add form 1299 api, model and handlers

**Description**

Adds API endpoint for shipment and storage application for PPP (https://drive.google.com/drive/u/0/search?q=1299) in API via swagger yaml. Currently has a single field of type string, "name_of_preparing_office". The schema needs to be architected and a proposal laid out before continuing with column definitions.

Adds ShipmentApplication model and migration and tests for the handlers.

**Testing Steps**

1. Run migrations via: `make db_dev_migrate`

2. With server running, run GET and POST requests via:

GET:
		`curl http://localhost:8080/api/v1/shipmentapplications`

POST: 
                `curl http://localhost:8080/api/v1/shipmentapplications --data "{ \"name_of_preparing_office\": \"pumpernickle office\"}"`

...which should return something like this:

`{"id":"7fb7d858-d3b6-489a-a6b2-8130fda7639d","created_at":"2018-01-18T23:20:40.627410118-08:00","updated_at":"2018-01-18T23:20:40.62741124-08:00","name_of_preparing_office":"sourdough office"}`

3. Run tests via: `make test`
